### ISSUE TITLE
Latest version doesn't save multiple rows in timesheet (wktime module)

### ISSUE THREAD

**Issue by aleroot** [dhanasingh/redmine_wktime#145]
When trying to save more than one row in the wktime module just one row is saved, and if you try to insert more than one row you get a 500 error due to a null pointer exception.

The stacktrace is : 

```
Completed 500 Internal Server Error in 333ms (ActiveRecord: 147.4ms)

NoMethodError (undefined method `each_with_index' for nil:NilClass):
  plugins/redmine_wktime/app/controllers/wktime_controller.rb:1193:in `block in gatherEntries'
  plugins/redmine_wktime/app/controllers/wktime_controller.rb:1177:in `each'
  plugins/redmine_wktime/app/controllers/wktime_controller.rb:1177:in `each_with_index'
  plugins/redmine_wktime/app/controllers/wktime_controller.rb:1177:in `gatherEntries'
  plugins/redmine_wktime/app/controllers/wktime_controller.rb:172:in `update'
  lib/redmine/sudo_mode.rb:63:in `sudo_mode'
```

The problem seems to be caused by a negative index in the row index variable.

**Comment by aleroot** [comment]
Opened the pull request #146 that fixes the issue, test on my local env and it is working fine.

**Comment by aleroot** [comment]
Ubuntu 16.04.4 LTS (GNU/Linux 4.4.0-122-generic x86_64)
ruby 2.3.1p112 (2016-04-26) [x86_64-linux-gnu]
erpmine 3.2.1

Output of gem list: 

```
*** LOCAL GEMS ***

actionmailer (5.2.0, 4.2.8)
actionpack (5.2.0, 4.2.8)
actionpack-action_caching (1.2.0)
actionpack-xml_parser (2.0.1, 1.0.2)
actionview (5.2.0, 4.2.8)
activejob (5.2.0, 4.2.8)
activemodel (5.2.0, 4.2.8)
activerecord (5.2.0, 4.2.8)
activesupport (5.2.0, 4.2.8)
addressable (2.5.2)
arel (9.0.0, 6.0.4)
awesome_nested_set (3.1.4)
bigdecimal (1.2.8)
builder (3.2.3)
bundler (1.16.1)
coderay (1.1.2)
coffee-rails (4.2.2)
coffee-script-source (1.12.2)
concurrent-ruby (1.0.5)
connection_pool (2.2.1)
crass (1.0.4)
css_parser (1.6.0)
did_you_mean (1.0.0)
erubi (1.7.1)
erubis (2.7.0)
execjs (2.7.0)
globalid (0.4.1)
hike (2.1.3)
htmlentities (4.3.4)
i18n (1.0.1, 0.7.0)
io-console (0.4.5)
jbuilder (2.7.0)
jquery-rails (4.3.3, 3.1.5)
json (1.8.3)
listen (3.1.5)
loofah (2.2.2)
mail (2.7.0, 2.6.6)
method_source (0.9.0)
mime-types (3.1)
mime-types-data (3.2016.0521)
mimemagic (0.3.2)
mini_mime (1.0.0)
mini_portile2 (2.3.0)
minitest (5.11.3, 5.8.4)
molinillo (0.6.5)
multi_json (1.13.1)
mysql2 (0.4.10)
net-http-persistent (3.0.0)
net-ldap (0.16.1, 0.12.1)
net-telnet (0.1.1)
nokogiri (1.8.2)
polyglot (0.3.5)
power_assert (1.1.1, 0.2.7)
protected_attributes (1.1.4)
psych (2.0.17)
public_suffix (3.0.2)
rack (2.0.5, 1.6.10, 1.6.4)
rack-openid (1.4.2)
rack-test (1.0.0, 0.6.3)
rails (4.2.8)
rails-deprecated_sanitizer (1.0.3)
rails-dom-testing (2.0.3, 1.0.9)
rails-html-sanitizer (1.0.4)
rails-observers (0.1.5)
railties (5.2.0, 4.2.8)
rake (12.3.1, 10.5.0)
rb-fsevent (0.10.3)
rb-inotify (0.9.10)
rbpdf (1.19.3)
rbpdf-font (1.19.1, 1.19.0)
rdoc (6.0.3, 4.2.1)
redcarpet (3.4.0)
request_store (1.4.1, 1.0.5)
rmagick (2.16.0)
roadie (3.3.0, 3.2.2)
roadie-rails (1.3.0, 1.1.1)
ruby-openid (2.3.0)
ruby_dep (1.5.0)
rufus-scheduler (3.0.3)
sass (3.5.6)
sass-listen (4.0.0)
sass-rails (5.0.7)
sdoc (1.0.0)
spring (2.0.2)
sprockets (3.7.1)
sprockets-rails (3.2.1)
test-unit (3.2.7, 3.1.7)
thor (0.20.0)
thread_safe (0.3.6)
tilt (2.0.8)
treetop (1.6.10)
turbolinks (5.1.1)
turbolinks-source (5.1.0)
tzinfo (1.2.5)
uglifier (4.1.10)
```

**Comment by aleroot** [comment]
Pushed new changes to fix the issue, please someone review the Pull Request and merge...

**Comment by aleroot** [comment]
There are still times whereas the POST request comes without ids, see the following request the ids3 has no values and cause a NullPointerException.

`utf8=%E2%9C%93&authenticity_token=CtZhhXJQMJRAcAn%2BAK6OKZoAJHItMVYlLGP3X2JjWedHk%2FCnkrJhuWCNBufpNlL5uI2ygOf%2BZIwZtKSm2Rg%2BYA%3D%3D&tab=wktime&getissues_url=%2Fwktime%2Fgetissues&getactivities_url=%2Fwktime%2Fgetactivities&deleterow_url=%2Fwktime%2Fdeleterow&user_id=18&startday=2018-05-14&time_entry%5B%5D%5Bproject_id%5D=167&time_entry%5B%5D%5Bissue_id%5D=&time_entry%5B%5D%5Bactivity_id%5D=20&time_entry%5B%5D%5Bcomments%5D=&hours1%5B%5D=6.00&ids1%5B%5D=28490&disabled1%5B%5D=false&hours1%5B%5D=&ids1%5B%5D=&disabled1%5B%5D=false&hours1%5B%5D=&ids1%5B%5D=&disabled1%5B%5D=false&hours1%5B%5D=&ids1%5B%5D=&disabled1%5B%5D=false&hours1%5B%5D=&ids1%5B%5D=&disabled1%5B%5D=false&hours1%5B%5D=&ids1%5B%5D=&disabled1%5B%5D=false&hours1%5B%5D=&ids1%5B%5D=&disabled1%5B%5D=false&total=6.00&unit=&hidden_clock_in_out=&time_entry%5B%5D%5Bproject_id%5D=51&time_entry%5B%5D%5Bissue_id%5D=&time_entry%5B%5D%5Bactivity_id%5D=9&time_entry%5B%5D%5Bcomments%5D=&hours3%5B%5D=2&ids3%5B%5D=&disabled3%5B%5D=false&hours3%5B%5D=&ids3%5B%5D=&disabled3%5B%5D=false&hours3%5B%5D=&ids3%5B%5D=&disabled3%5B%5D=false&hours3%5B%5D=&ids3%5B%5D=&disabled3%5B%5D=false&hours3%5B%5D=&ids3%5B%5D=&disabled3%5B%5D=false&hours3%5B%5D=&ids3%5B%5D=&disabled3%5B%5D=false&hours3%5B%5D=&ids3%5B%5D=&disabled3%5B%5D=false&wktime_save=Salva
`

The exception thrown is: 
```

NoMethodError (undefined method `each_with_index' for nil:NilClass):
  plugins/redmine_wktime/app/controllers/wktime_controller.rb:1193:in `block in gatherEntries'
  plugins/redmine_wktime/app/controllers/wktime_controller.rb:1177:in `each'
  plugins/redmine_wktime/app/controllers/wktime_controller.rb:1177:in `each_with_index'
  plugins/redmine_wktime/app/controllers/wktime_controller.rb:1177:in `gatherEntries'
  plugins/redmine_wktime/app/controllers/wktime_controller.rb:172:in `update'
  lib/redmine/sudo_mode.rb:63:in `sudo_mode'
```

**Comment by aleroot** [comment]
I've fixed for our needs with the latest updated pull request commit, anyway the owner of the project should review the pull request and the entire module since it has a lot of errors in the edit.js Javascript file, and it seems like it hasn't been tested.

**Comment by dhanasingh** [comment]
thanks for the fix, the code is merged in version 3.3, please take a look
